server/workflowarchive/archived_workflow_server.go

```diff
@@ -188,3 +191,24 @@ func (w *archivedWorkflowServer) ListArchivedWorkflowLabelValues(ctx context.Con
 	}
 	return labels, err
 }
+
+func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req *workflowarchivepkg.RetryArchivedWorkflowRequest) (*wfv1.Workflow, error) {
+	wfClient := auth.GetWfClient(ctx)
+	kubeClient := auth.GetKubeClient(ctx)
+
+	wf, err := w.wfArchive.GetWorkflow(req.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	err = w.wfArchive.ValidateWorkflow(wf)
```

Comment:
this is correct - we should not retry workflows that belong to another instance - but is this check not redundant? GetWorkflow should never return a workflow which does not have correct instance ID

Reply:
This check comes from following the `RetryWorkflow` method and Bala and I discussed that it is redundant with the Get. In this case, we could remove `ValidateWorkflow` altogether for wfArchive also addressing the other issue that it doesn't belong with workflow archive.